### PR TITLE
Document Bag Logic

### PR DESCRIPTION
**Pull request by Fexty12573**
Seeing as #11 seems to be abandoned, I decided to do this instead.
Documents bag functions. Logic only. The bottom part of `bag.c` constains data related to visuals so I didn't document them. Gonna do that in another PR.

### CODE REVIEW

File: src/bag.c [hunk 1]
```diff
@@ -0,0 +1,490 @@
...
+        pocketSize = TMHM_POCKET_SIZE;
+        break;
+    }
+
+    return slot >= pocketSize ? NULL : &pocket[slot];
+}
+
+Bag *SaveData_GetBag(SaveData *saveData)
+{
+    return SaveData_SaveTable(saveData, 3);
```
Comment (lhearachel) [inline src/bag.c:408]: **polish:** there should be an enum value in `constants/savedata/save_table.h` that you can use here

File: include/bag.h [hunk 2]
```diff
@@ -0,0 +1,85 @@
+#ifndef POKEPLATINUM_BAG_H
+#define POKEPLATINUM_BAG_H
+
+#include "struct_defs/struct_0207CDEC.h"
+#include "bag.h"
+#include "bag.h"
```
[the remark this answers is not in the document]
Comment (Fexty12573) [reply include/bag.h:6]: Whoops, that happened when I find-and-replaced all struct decl includes with bag.h, seems I forgot to fix this one 😆

File: src/bag.c [hunk 3]
```diff
@@ -0,0 +1,490 @@
...
+static BagItem *Pocket_FindSlotToRemoveItem(BagItem *pocket, u32 pocketSize, u16 item, u16 count)
+{
+    for (u32 i = 0; i < pocketSize; i++) {
+        if (pocket[i].item == item) {
+            return pocket[i].quantity >= count ? &pocket[i] : NULL;
+        }
+    }
+
+    return NULL;
+}
```
[the remark this answers is not in the document]
Comment (Fexty12573) [reply src/bag.c:170]: Makes sense, I mainly picked that name as a counterpart to `Bag_FindSLotToAddItem` but I don't mind changing it.

**Review by lhearachel — APPROVED** [review]
Will merge once conflicts are addressed.

**Pull request merged**